### ISSUE TITLE
Error processing the file (big file)

### ISSUE THREAD

**Issue by wardpeet** [GoogleChrome/big-rig#28]
Cli works fine but webapp keeps throwing Error processing the file.
The timeline.json file is 12.8mb. When i upload aerotwist.json it works fine. Probably because the file it soo big?

**Comment by paullewis** [comment]
There's generally a problem with Python's JSON parsing when it comes to _some_ of the trace files :-/ I'm still trying to figure it out, but if you'd be willing to post the file somewhere I'll gladly take a look and see if I can figure out what's going on. Did you take the file from WebPagetest or Timeline?

**Comment by wardpeet** [comment]
Here you go http://www.webpagetest.org/result/151215_9S_1aa13ab94c8e695ee2b30666781d362c/

**Comment by axemclion** [comment]
I had similar problem when I was working with [browser-perf](http://github.com/axemclion/browser-perf). The tracing files can get really big and the way I solved this was by reading the files in streams. This is just an array of records, and parsing the file as a stream let me parse.

node-big-rig could also benefit from reading the stream instead of reading the [entire big file](https://github.com/GoogleChrome/node-big-rig/blob/58abe93434ffda015c8d8edf00b2d8f04cc2a211/bigrig.js#L58)

**Comment by datrinh** [comment]
I have the same problem. It says `JSON parse error`
My file is 25 MB tho :/ Link to json file: https://www.dropbox.com/s/ui5udnmv96jqjb7/timeline_single.json?dl=0

**Comment by axemclion** [comment]
@datrinh Did you try using http://github.com/axemclion/browser-perf/ ?

**Comment by datrinh** [comment]
@axemclion No, I haven't. I didn't plan on setting up multiple tools if possible tho :)

**Comment by datrinh** [comment]
I'm still having the error even on smaller files (5 MB)...
Is there any alternative app that I can use until this is fixed?

**Comment by zzarcon** [comment]
Same issue for me, file size <7mb. I also tried disabling/enabling Timeline options like `Js profile`, `Memory`, `Paint`, `Screeshots`, etc. And still doesn't work.
